src/screens/Watch/Components/Transcriptions/CaptionLine/index.js

```diff
@@ -1,140 +1,233 @@
-      }
-    }
-  };
+  const [fullBeginTime, setFullBeginTime] = useState(begin);
+  const [fullEndTime, setFullEndTime] = useState(end);
 
-  const handleSeek = () => {
-    const time = timeStrToSec(begin);
-    dispatch({ type: 'watch/media_setCurrTime', payload: time })
-  };
+  const [timeString, setTimeString] = useState(prettierTimeStr(begin));
+  const [endTimeString, setEndTimeString] = useState(prettierTimeStr(end));
 
-  const handleChange = () => {
-    // console.log(target.innerText)
-  };
+  useEffect(() => {
+    // eslint-disable-next-line no-console
+    console.log("Full Begin Time Updated:", fullBeginTime);
+  }, [fullBeginTime]);
 
-  const handleFocus = ({ target }) => {
-    // console.error(e.target.innerText)
-    dispatch({ type: 'watch/setTransEditMode', payload: { caption, innerText: target.innerText } })
+  useEffect(() => {
+    // eslint-disable-next-line no-console
+    console.log("Full End Time Updated:", fullEndTime);
+  }, [fullEndTime]);
+
+  const validateTimeFormat = (input) => {
+    const timeRegex = /^(\d{1,2}:)?\d{1,2}:\d{2}(\.\d+)?$/;
+    if (!timeRegex.test(input)) {
+      throw new Error('Invalid time format');
+    }
+    return true;
   };
 
-  const handleBlur = () => {
-    ref.current.innerText = text;
-    transControl.handleBlur();
+  const handleCancel = () => {
+    textRef.current.innerText = text;
+    timeRef.current.innerText = prettierTimeStr(fullBeginTime);
+    endTimeRef.current.innerText = prettierTimeStr(fullEndTime);
+    dispatch({ type: 'watch/setCurrEditing', payload: null });
   };
 
-  const handleSave = () => {
-    dispatch({ type: 'watch/saveCaption', payload: { caption, text: ref.current.innerText } })
+  const handleSave = (updatedBeginTime = fullBeginTime, updatedEndTime = fullEndTime) => {
+    const newText = textRef.current.innerText;
+  
+    try {
+      // eslint-disable-next-line no-console
+      console.log("Handle save triggered");
+      // eslint-disable-next-line no-console
+      console.log("Using begin time:", updatedBeginTime);
+      // eslint-disable-next-line no-console
+      console.log("Using end time:", updatedEndTime);
+  
+      validateTimeFormat(updatedBeginTime);
+      validateTimeFormat(updatedEndTime);
+      dispatch({
+        type: 'watch/saveCaption',
+        payload: { caption, text: newText, begin: updatedBeginTime, end: updatedEndTime },
+      });
+      setTimeString(prettierTimeStr(updatedBeginTime));
+      setEndTimeString(prettierTimeStr(updatedEndTime));
+      // eslint-disable-next-line no-console
+      console.log("Time strings updated to:", prettierTimeStr(updatedBeginTime), prettierTimeStr(updatedEndTime));
+      textRef.current.innerText = newText;
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error("Error during save:", error.message);
+      alert('Please enter a valid time format (HH:MM:SS, MM:SS, or HH:MM:SS.SSS)');  // eslint-disable-line no-alert

```

Comment:
Please use a better alert library (ClassTranscribe already has a couple included) than the js builtin 'alert'